Rover/mode_smart_rtl.cpp

```diff
@@ -73,23 +79,31 @@ void ModeSmartRTL::update()
             }
             break;
 
+        case SmartRTL_Loiter:
+            if(_reached_destination)
+                rover.mode_loiter.update();
+            else {
+                if (g2.smart_rtl.request_thorough_cleanup()) {
+                    smart_rtl_state = SmartRTL_PathFollow;
+                    _load_point = true;
+                    break;
+                } else {
+                    rover.mode_loiter.update();
+                }
+            }
```

Comment:
This can all me simplified; essentially one break or a `rover.mode_loiter.update()` - and certainly fewer `else`s